server/bundler/remove-parent-modules-monkeypatch.js

```diff
@@ -0,0 +1,116 @@
+// this is really gross.
+//
+// originally from
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+"use strict"; //eslint-disable-line
+
+var forEach = require( 'lodash/forEach' ),
+	filter = require( 'lodash/filter' ),
+	indexOf = require( 'lodash/indexOf' ),
+	pullAll = require( 'lodash/pullAll' ),
+	nanotimer = require( 'nanotimer' );
+
+var RemoveParentModulesPlugin = require( 'webpack/lib/optimize/RemoveParentModulesPlugin' );
+var Chunk = require( 'webpack/lib/Chunk' );
+var Module = require( 'webpack/lib/Module' );
+
+module.exports = RemoveParentModulesPlugin;
+
+function hasModule( chunk, module, checkedChunks ) {
+	if ( ~ indexOf( chunk.modules, module ) ) {
+		return [chunk];
+	}
+	if ( chunk.entry ) {
+		return false;
+	}
+	const uncheckParents = filter( chunk.parents, function( c ) {
+		return ! checkedChunks.has( c );
+	} );
+	return allHaveModule( uncheckParents, module, checkedChunks );
+}
+
+function allHaveModule( someChunks, module, checkedChunks ) {
+	if ( ! checkedChunks ) {
+		checkedChunks = new Set();
+	}
+	let chunks = new Set();
+	for ( let i = 0; i < someChunks.length; i++ ) {
+		checkedChunks.add( someChunks[i] );
+		const subChunks = hasModule( someChunks[i], module, checkedChunks );
+		if ( ! subChunks ) {
+			return false;
+		}
+		addToSet( chunks, subChunks );
+	}
+	return chunks;
+}
+
+function addToSet( set, items ) {
+	forEach( items, function( item ) {
+		set.add( item );
+	} );
+}
+
+Chunk.prototype._calypso_removeModules = function( modules ) {
+	this.modules = pullAll( this.modules, modules );
+	forEach( modules, function( module ) {
+		module._calypso_removeChunk( this );
+	} );
+}
+
+const oldAddParent = Chunk.prototype.addParent;
+Chunk.prototype.addParent = function( parent ) {
+	console.log( 'Adding %s as a parent of %s', parent.name, this.name );
+	return oldAddParent.apply( this, arguments );
+}
+
+Module.prototype._calypso_removeChunk = function( chunk ) {
```

Comment:
Module calls back into chunk to remove itself, but in this case, we know that we're already gone from the calling chunk, so avoid it. Avoids a linear search of modules, which can get pretty big.